execution/src/windows/execution.cc

```diff
@@ -117,11 +118,11 @@ namespace leatherman { namespace execution {
 
         // Format a name for the pipe based on the process and counter
         wstring name = boost::nowide::widen((boost::format("\\\\.\\Pipe\\leatherman.%1%.%2%") %
-            GetCurrentProcessId() %
+            GetCurrentThreadId() %
```

Comment:
I suppose there's a scenario using timeouts where we could release this end of the named pipe but the other end doesn't release. Then this thread shuts down and another with the same thread id is started and reconnects to the existing named pipe. That becomes less likely with both process id and thread id (we could throw in a random number as well just to be safe).